### PR TITLE
Add failure reason input on failed task screen

### PR DESCRIPTION
**Pull request by r0xsh**
image

Fixes: https://github.com/coopcycle/coopcycle-app/issues/1546

### CODE REVIEW

**Comment by r0xsh** [comment]
Depends on: coopcycle/coopcycle-web#3792

**Comment by r0xsh** [comment]
Fixes: #1546

File: src/navigation/task/Complete.js [hunk 1]
```diff
@@ -53,19 +76,54 @@ const AttachmentItem = ({ base64, onPressDelete }) => {
   )
 }
 
+const FailureReasonPicker = ({ task, httpClient, onValueChange }) => {
+
+  const { data, isSuccess } = useQuery([ 'task', 'failure_reasons', task['@id'] ], async () => {
+    return await httpClient.get(`${task['@id']}/failure_reasons`)
```
Comment (alexsegura) [inline src/navigation/task/Complete.js:82]: To be compatible with older servers, when this endpoint returns a 404, the reason picker shouldn't appear.

Comment (r0xsh) [reply src/navigation/task/Complete.js:84]: Tried to hide it if 404 is returns but it was looking clunky, so i preferred to add an error message instead
image

**Pull request merged**